internal/metrics/metrics.go

```diff
@@ -0,0 +1,89 @@
+package metrics
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	registry *prometheus.Registry
+	once     sync.Once
+
+	// BackendLatency measures request duration for LLM backend calls
+	BackendLatency *prometheus.HistogramVec
+
+	// TokensTotal tracks token usage per model and type
+	TokensTotal *prometheus.CounterVec
+
+	// RequestsTotal tracks total requests by backend, model and status
+	RequestsTotal *prometheus.CounterVec
+
+	// FirstTokenLatency measures time to first token in streaming responses
+	FirstTokenLatency *prometheus.HistogramVec
+
+	// InterTokenLatency measures time between tokens in streaming responses
+	InterTokenLatency *prometheus.HistogramVec
```

Comment:
let's avoid the global state, but instantiate them in the main routine and pass them through via a struct field for testability.

Reply:
sounds good, will follow up later to get these done